contracts/LockRelease.sol

```diff
@@ -1,101 +1,162 @@
-    uint128 public duration; // duration of the release schedule in seconds
-
-    address[] public beneficiaries; // list of beneficiaries
-
     /** Represents a release schedule for a specific beneficiary. */
     struct Schedule {
         uint256 total; // total tokens that the beneficiary will receive over the duration
         uint256 released; // already released tokens to the beneficiary
     }
 
+    address public token; // address of the token being released
+    uint128 public start; // start timestamp of the release schedule
+    uint128 public duration; // duration of the release schedule in seconds
+
+    address[] private beneficiaries; // list of beneficiaries
+
     /** Represents a release schedule for a specific beneficiary. */
     mapping(address => Schedule) private schedules;
 
     /** Emitted when a release schedule is created. */
-    event ScheduleStarted(uint256 total, uint128 start, uint128 duration);
+    event ScheduleStarted(
+        address token,
+        address[] beneficiaries,
+        uint256[] amounts,
+        uint128 start,
+        uint128 duration
+    );
 
     /** Emitted when tokens are released to a recipient. */
-    event TokensReleased(
-        address indexed beneficiary,
-        uint256 amount,
-        address indexed releasor
-    );
+    event TokensReleased(address indexed beneficiary, uint256 amount);
 
-    error InvalidInputs();
+    error InvalidArrayLengths();
     error ZeroDuration();
     error InvalidBeneficiary();
     error InvalidToken();
     error InvalidAmount();
     error DuplicateBeneficiary();
     error NothingToRelease();
 
-    constructor(address _token, address[] memory _beneficiaries, uint256[] memory _amounts, uint128 _start, uint128 _duration) {
-        
-        if (token == address(0)) revert InvalidToken();
-        if (duration == 0) revert ZeroDuration();
-        if (_beneficiaries.length != _amounts.length) revert InvalidInputs();
-
-        token = _token;
-        start = _start;
-        duration = _duration;
-
-        uint256 total = 0;
-
-        for (uint16 i = 0; i < _beneficiaries.length; i++) {
+    /** Deploys the LockRelease contract and sets up all beneficiary release schedules.
+     *
+     * @param _token address of the beneficiary
+     * @param _beneficiaries array of beneficiary addresses to create release schedules for
+     * @param _amounts array of the amount of tokens to be locked and released for each beneficiary
+     * @param _start the timestamp that the release schedule begins releasing tokens at
+     * @param _duration the time period in seconds that tokens are released over
+     */
+    constructor(
+        address _token,
+        address[] memory _beneficiaries,
+        uint256[] memory _amounts,
+        uint128 _start,
+        uint128 _duration
+    ) EIP712("DecentLockRelease", "1.0.0") {
+        if (_token == address(0)) revert InvalidToken();
+        if (_duration == 0) revert ZeroDuration();
```

Comment:
fair enough, yeah, though I think it might instill more trust for folks looking into our contracts

Reply:
Just tested this, removing all the revert checks from the constructor decreases cost of deploying the contract by 0.2%. Since its negligible I'm thinking we just leave them in